include/nbl/asset/utils/shaderCompiler_serialization.h

```diff
@@ -0,0 +1,200 @@
+#ifndef  _NBL_ASSET_SHADER_COMPILER_SERIALIZATION_H_INCLUDED_
+#define _NBL_ASSET_SHADER_COMPILER_SERIALIZATION_H_INCLUDED_
+
+#include "nbl/asset/utils/IShaderCompiler.h"
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+using SEntry = nbl::asset::IShaderCompiler::CCache::SEntry;
+
+
```

Comment:
this can all do into `src` not `include`

Nobody should care about how we serialize our cache.